packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -565,6 +565,34 @@ the configured path by running this command: flutter config --android-studio-dir
     }
   }
 
+  static Version? _parseVersion(String? text) {
+    Match? match;
+    // Example string fort EAP: EAP AI-242.21829.142.2422.12358220
+    // We try to capture 2422 here.
+    final RegExp intellijPattern = RegExp(r'EAP\s+[A-Z]{2}-\d+\.\d+\.\d+\.(\d+)\.\d+');
+    match = intellijPattern.firstMatch(text ?? '');
+
+    if (match == null) {
+      return Version.parse(text);
+    }
+
+    final String? matched = match.group(1);
```

Comment:
I have a couple of notes here:

* I don't see a benefit in declaring `match` before it's needed. The name of the function (`_parseVersion`) and existence of a regex pattern already provide a good hint that we are going to be doing some regex-parsing here.
* I am not sure how accurate `intellijPattern` is as a variable name. Also, I think `matched` could be named to something more descriptive.

Consider:

```dart
    // Matches the version string for Preview builds on macOS.
    // Example match: EAP AI-242.21829.142.2422.12358220
    // We try to capture "2422" here, which can be translated to a
    // more human-friendly "24.2.2".
    final RegExp eapVersionPattern = RegExp(
      r'EAP\s+[A-Z]{2}-\d+\.\d+\.\d+\.(\d+)\.\d+',
    );

    final Match? eapVersionMatch = eapVersionPattern.firstMatch(text);

    if (eapVersionMatch == null) {
      return Version.parse(text);
    }

    // This would be "2422" in the earlier example.
    final String? normalVersionPart = eapVersionMatch.group(1);
```

Reply:
Thanks! Played a little with naming here. And eliminated separate declaration of `match` – it's agree that there's no benefit.